test/test_model.py

```diff
@@ -65,202 +71,230 @@ def setUp(self):
             secretfoo:
                 type: secret
         ''')
-        self.addCleanup(self.harness.cleanup)
-        self.relation_id_db0 = self.harness.add_relation('db0', 'db')
-        self.harness._get_backend_calls(reset=True)
-        self.model = self.harness.model
+        yield harness
+        harness.cleanup()
+
+    @pytest.fixture
+    def model(self, harness: ops.testing.Harness[ops.CharmBase]):
+        return harness.model
+
+    @pytest.fixture
+    def relation_id_db0(self, harness: ops.testing.Harness[ops.CharmBase]):
```

Comment:
I think it'd be simpler if this were just done explicitly in tests that need it, or a helper function called explicitly by tests that need it. And it's kind of unclear what the fixture means/does.